### PR TITLE
feat(api): #MC-161 add trace to api

### PR DESCRIPTION
**Pull request by AyaNADINI**
## Describe your changes
Providing a way to determine what exactly is happening with HTTP request, to diagnose unexpected or undesirable behavior.
To deal with this, we add API trace to trace HTTP request on specific routes.
Trace annotation contains two parameters :
value (String): Set action name
body (Boolean): Set if the request body should be saved(by default body takes true)

## Checklist tests
Create a ressource in rbs, and then check in mongodb using `db.getCollection('traces').find({"action":"CREATE_RESOURCE_TYPE"})`
 

## Issue ticket number and link
MC-161 : https://entsupport.gdapublic.fr/browse/MC-161

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by qpic-cgi — APPROVED** [review]

File: src/main/java/net/atos/entng/rbs/core/constants/Actions.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.entcore.common.http.filter.Trace;
+
+public class Actions {
+    private Actions() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    // Trace actions
+
+    public static final String CREATE_AVAILABILITY= "CREATE_AVAILABILITY";
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/rbs/core/constants/Actions.java]: indentation =, il faut des espaces

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/java/net/atos/entng/rbs/controllers/ResourceTypeController.java [hunk 2]
```diff
@@ -261,6 +266,7 @@ public void shareJson(final HttpServerRequest request){
 	@Put("/share/json/:id")
 	@ApiDoc("Add rights for a given resource type")
 	@SecuredAction(value = "rbs.manager", type = ActionType.RESOURCE)
+	@Trace(Actions.SHARE_JSON_SUBMIT)
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/rbs/controllers/ResourceTypeController.java]: SHARE_RESOURCE_TYPE_SUBMIT

File: src/main/java/net/atos/entng/rbs/controllers/ResourceTypeController.java [hunk 3]
```diff
@@ -270,6 +276,7 @@ public void shareJsonSubmit(final HttpServerRequest request){
 	@Put("/share/remove/:id")
 	@ApiDoc("Remove rights for a given resource type")
 	@SecuredAction(value = "rbs.manager", type = ActionType.RESOURCE)
+	@Trace(Actions.REMOVE_SHARE)
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/rbs/controllers/ResourceTypeController.java]: SHARE_RESOURCE_TYPE_REMOVE

File: src/main/java/net/atos/entng/rbs/controllers/ResourceTypeController.java [hunk 4]
```diff
@@ -345,6 +354,7 @@ public void handle(final UserInfos user) {
 	}
 	@Put("/share/resource/:id")
 	@SecuredAction(value = "rbs.manager", type = ActionType.RESOURCE)
+	@Trace(Actions.SHARE_RESOURCE)
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/rbs/controllers/ResourceTypeController.java]: SHARE_RESOURCE_TYPE

File: src/main/java/net/atos/entng/rbs/controllers/ResourceController.java [hunk 5]
```diff
@@ -391,6 +398,7 @@ public void share(final HttpServerRequest request) {
 	@ApiDoc("Add rights for a given resource")
 	@ResourceFilter(TypeAndResourceAppendPolicy.class)
 	@SecuredAction(value = "rbs.manager", type = ActionType.RESOURCE)
+	@Trace(Actions.SHARE_SUBMIT)
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/rbs/controllers/ResourceController.java]: SHARE_RESOURCE_SUBMIT

File: src/main/java/net/atos/entng/rbs/controllers/ResourceController.java [hunk 6]
```diff
@@ -399,6 +407,7 @@ public void shareSubmit(final HttpServerRequest request) {
 	@ApiDoc("Remove rights for a given resource")
 	@ResourceFilter(TypeAndResourceAppendPolicy.class)
 	@SecuredAction(value = "rbs.manager", type = ActionType.RESOURCE)
+	@Trace(Actions.SHARE_REMOVE)
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/rbs/controllers/ResourceController.java]: SHARE_RESOURCE_REMOVE

**Review by alicedraillard — APPROVED** [review]

**Review by JinDouang — APPROVED** [review]

**Review by alicedraillard — APPROVED** [review]

**Pull request merged**